packages/ledger/src/lib/ledger.ts

```diff
@@ -252,9 +253,14 @@ const Ledger: WalletBehaviourFactory<HardwareWallet> = async ({
         options.network
       );
 
-      return Promise.all(
-        signedTransactions.map((signedTx) => provider.sendTransaction(signedTx))
-      );
+      const results: Array<FinalExecutionOutcome> = [];
+
+      for (let i = 0; i < signedTransactions.length; i++) {
+        const tsx = await provider.sendTransaction(signedTransactions[i]);
+        results.push(tsx);
```

Comment:
I'd suggest refactoring this to simply:
```ts
results.push(await provider.sendTransaction(signedTransactions[i]));
```
Or renaming `tsx` to `result` as there's a couple of things wrong with this name:
- It's the "outcome" (or result) of the transaction and not the transaction itself
- We generally use the name `tx` or `transaction` when referring to a transaction